src/client/utils/drag-manager/drag-manager.ts

```diff
@@ -31,24 +30,26 @@ interface DraggedElementBase<T> {
 
 interface DraggedDimension extends DraggedElementBase<Dimension> {
   type: DraggedElementType.DIMENSION;
-  origin: DimensionOrigin;
 }
 
 interface DraggedMeasure extends DraggedElementBase<Measure> {
   type: DraggedElementType.MEASURE;
-  origin: MeasureOrigin;
 }
 
-type DraggedElement = DraggedDimension | DraggedMeasure;
+interface DraggedSeries extends DraggedElementBase<Series> {
+  type: DraggedElementType.SERIES;
+}
 
-function isDimension(el: DraggedElement): el is DraggedDimension {
-  return el.type === DraggedElementType.DIMENSION;
+interface DraggedSplit extends DraggedElementBase<Split> {
+  type: DraggedElementType.SPLIT;
 }
 
-function isMeasure(el: DraggedElement): el is DraggedMeasure {
-  return el.type === DraggedElementType.MEASURE;
+interface DraggedFilter extends DraggedElementBase<FilterClause> {
+  type: DraggedElementType.FILTER;
 }
 
+type DraggedElement = DraggedDimension | DraggedMeasure | DraggedFilter | DraggedSplit | DraggedSeries;
+
 export class DragManager {
```

Comment:
All methods throws NPE if dragging is not set.

Reply:
Resolved in 9a8c3e1